mlflow/server/js/src/components/MetricsPlotView.js

```diff
@@ -1,13 +1,25 @@
 import React from 'react';
 import Utils from '../utils/Utils';
 import _ from 'lodash';
-import PropTypes from 'prop-types';
+import PropTypes, {array} from 'prop-types';
 import { X_AXIS_STEP, X_AXIS_RELATIVE } from './MetricsPlotControls';
 import { CHART_TYPE_BAR } from './MetricsPlotPanel';
 import Plot from 'react-plotly.js';
 
 const MAX_RUN_NAME_DISPLAY_LENGTH = 36;
 
+function EMA(mArray, mRange) {
+  const k = 2 / (mRange + 1);
```

Comment:
@Valentyn1997 I tried out this EMA code and found that the behavior is a little bit counterintuitive in certain cases. I've gone ahead and prototyped an alternative routine inspired by TensorBoard. Here are some examples of the before (the EMA routine introduced by this PR) and after (the prototyped EMA routine):

Before:
![before1](https://user-images.githubusercontent.com/39497902/77599562-5959f580-6ec2-11ea-9a93-1bb48e33c1db.gif)
After:
![after1](https://user-images.githubusercontent.com/39497902/77599568-60810380-6ec2-11ea-93b8-649016bdf91f.gif)

Before:
![before2](https://user-images.githubusercontent.com/39497902/77599576-6676e480-6ec2-11ea-8c9e-40c0a1f5f912.gif)
After:
![after2](https://user-images.githubusercontent.com/39497902/77599579-6a0a6b80-6ec2-11ea-9301-d9dccf8fcbce.gif)

Before:
![before3](https://user-images.githubusercontent.com/39497902/77599584-6ecf1f80-6ec2-11ea-8c54-e04566d32517.gif)
After:
![after3](https://user-images.githubusercontent.com/39497902/77599592-74c50080-6ec2-11ea-88f8-a5dceaa9a552.gif)

In these three cases, I believe that the `after` examples provide more accurate smoothing behavior. Here's the proposed routine, which performs a bias correction ("debiasing") routine to ensure that the first few elements of the metric value set aren't artificially skewed lower:

```
function EMA(mArray, smoothingWeight) {
  // If all elements in the set of metric values are constant, or if
  // the degree of smoothing is set to the minimum value, return the
  // original set of metric values
  if (smoothingWeight <= 1 || mArray.every((v) => v === mArray[0])) {
    return mArray;
  }

  const smoothness = smoothingWeight / 101;
  const smoothedArray = [];
  let biasedElement = 0;
  for (let i = 0; i < mArray.length; i++) {
    biasedElement = (biasedElement * smoothness) + ((1 - smoothness) * mArray[i]);
    // To avoid biasing earlier elements toward smaller-than-accurate values, we divide
    // all elements by a `debiasedWeight` that asymptotically increases and approaches
    // 1 as the element index increases
    const debiasWeight = 1.0 - Math.pow(smoothness, i + 1);
    const debiasedElement = biasedElement / debiasWeight;
    smoothedArray.push(debiasedElement);
  }
  return smoothedArray;
}
```

Please let me know how you feel about using this routine instead (we may also want to replace the constant `101` with `max + 1` where `max` is the maximum value configured in the smoothness slider).

cc @smurching and @Zangr for a review here as well.

Reply:
Makes sense / looks reasonable to me!

Reply:
Yes, debiased version is much better!

Reply:
Added - thanks for the function code!